### ISSUE TITLE
Stretching overscroll with RTl Stretches in incorrect direction

### ISSUE THREAD

**Issue by youssefali424** [flutter/flutter#113185]
## Steps to Reproduce

1. Add rtl support to flutter app

**Expected results:** Scroll stretch to be same as native android and stretch effect stretches towards the drag position 

**Actual results:** stretch effect stretches towards opposite of the drag position 

<details>
<summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      theme: ThemeData.dark().copyWith(
        androidOverscrollIndicator: AndroidOverscrollIndicator.stretch,
      ),
      debugShowCheckedModeBanner: false,
      home: const Directionality(
        textDirection: TextDirection.rtl,
        // textDirection: TextDirection.ltr,
        child: Scaffold(
          body: SafeArea(
            child: Center(
              child: MyWidget(),
            ),
          ),
        ),
      ),
    );
  }
}

class MyWidget extends StatelessWidget {
  const MyWidget({super.key});

  @override
  Widget build(BuildContext context) {
    return SizedBox(
      height: 100,
      child: ListView.builder(
        itemBuilder: (context, index) => Container(
          width: 50,
          height: 100,
          color: Colors.amber[(index % 9 + 1) * 100],
          child: Center(child: Text('Entry $index')),
        ),
        itemCount: 10,
        padding: const EdgeInsets.all(8),
        scrollDirection: Axis.horizontal,
      ),
    );
  }
}
```

</details>

here two different videos when rtl support is added and another when ltr support is used

**RTL 👇**

https://user-images.githubusercontent.com/38167511/194758455-7a52dd6f-9623-4b27-9316-b8e54f6e0019.mp4

**LTR 👇**

https://user-images.githubusercontent.com/38167511/194758462-43335688-ceb1-4f71-94c9-afdb72683de4.mp4

**Comment by exaby73** [comment]
## Triage report

I can reproduce this issue. I have removed the deprecated use of `androidOverscrollIndicator` from `ThemeData` and instead used `useMaterial3`, but the issue still persists.

Adding a custom `ScrollBehavior` with the `axisDirection` set to `AxisDirection.right` fixes the issue.

<details>
<summary>Code Sample updated for the fix</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      theme: ThemeData.dark().copyWith(),
      debugShowCheckedModeBanner: false,
      home: const Directionality(
        textDirection: TextDirection.rtl,
        // textDirection: TextDirection.ltr,
        child: Scaffold(
          body: SafeArea(
            child: Center(
              child: MyWidget(),
            ),
          ),
        ),
      ),
    );
  }
}

class MyWidget extends StatelessWidget {
  const MyWidget({super.key});

  @override
  Widget build(BuildContext context) {
    return SizedBox(
      height: 100,
      child: ScrollConfiguration(
        behavior: MyBehaviour(),
        child: ListView.builder(
          itemBuilder: (context, index) => Container(
            width: 50,
            height: 100,
            color: Colors.amber[(index % 9 + 1) * 100],
            child: Center(child: Text('Entry $index')),
          ),
          itemCount: 10,
          padding: const EdgeInsets.all(8),
          scrollDirection: Axis.horizontal,
        ),
      ),
    );
  }
}

class MyBehaviour extends ScrollBehavior {
  @override
  Widget buildOverscrollIndicator(
      BuildContext context, Widget child, ScrollableDetails details) {
    return StretchingOverscrollIndicator(
      axisDirection: AxisDirection.right,
      child: child,
    );
  }
}
```

</details>

**Comment by youssefali424** [comment]
@exaby73 after further investigation now i know where is the issue here
<details>
<summary>Code sample</summary>

```dart
AlignmentDirectional _getAlignmentForAxisDirection(double overscroll) {
    // Accounts for reversed scrollables by checking the AxisDirection
    switch (widget.axisDirection) {
      case AxisDirection.up:
        return overscroll > 0
            ? AlignmentDirectional.topCenter
            : AlignmentDirectional.bottomCenter;
      case AxisDirection.right:
        return overscroll > 0
            ? AlignmentDirectional.centerEnd
            : AlignmentDirectional.centerStart;
      case AxisDirection.down:
        return overscroll > 0
            ? AlignmentDirectional.bottomCenter
            : AlignmentDirectional.topCenter;
      case AxisDirection.left:
        return overscroll > 0
            ? AlignmentDirectional.centerStart
            : AlignmentDirectional.centerEnd;
    }
  }
```

</details>
AlignmentDirectional is used but it is reversed with AxisDirection.left

<details>

<summary>Fix</summary>

```dart
AlignmentDirectional _getAlignmentForAxisDirection(double overscroll) {
    // Accounts for reversed scrollables by checking the AxisDirection
    switch (widget.axisDirection) {
      case AxisDirection.up:
        return overscroll > 0
            ? AlignmentDirectional.topCenter
            : AlignmentDirectional.bottomCenter;
      case AxisDirection.left:
      case AxisDirection.right:
        return overscroll > 0
            ? AlignmentDirectional.centerEnd
            : AlignmentDirectional.centerStart;
      case AxisDirection.down:
        return overscroll > 0
            ? AlignmentDirectional.bottomCenter
            : AlignmentDirectional.topCenter;
    }
  }
```

</details>

**Comment by youssefali424** [comment]
fix in #113214

**Comment by Piinks** [comment]
As stated above, this issue was fixed! 🎉
